determined-ai/determined
# feat: associate templates with workspaces

## Description

- break up `det template set` into `det template create|set-value`
- maintain old set behavior
- add an option to declarative argparse to mark options and commands as deprecated.
  -  warn on usage and on --help
- deprecate old template put endpoint and `det template set`


DET-9452 DET-9335




## Test Plan



- added crud tests for template apis

manual testing:
- create, update,  list, describe, and delete templates
- try creating with a specifying a specific workspace name and check the workspace id is persisted correctly. (check the api response for now or use det tpl ls)



## Commentary (optional)


maybe this is a another way we can updated the cli options:
https://github.com/determined-ai/determined/blob/81c57d06b42820c60053d25ec254a28cff1a4e41/harness/determined/cli/experiment.py#L1250-L1254

TODO:
- [x] api codes
- [x] depends on #6572 





## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket
